creator/data_templates/packaging.py

```diff
@@ -0,0 +1,272 @@
+from typing import Union, List
+import os
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+from io import BytesIO
+from pprint import pformat
+
+import pandas
+
+from creator.studies.models import Study
+from creator.data_templates.models import TemplateVersion
+
+MAX_SHEET_NAME_LEN = 31
+
+PACKAGE_INFO = (
+    """
+    The Fields file describes all of the fields or columns in the template
+    (name, data type, required or not, etc.) while the blank Template file
+    is what data submitters will fill out and then submit to their
+    organization for ingestion.
+    """
+)
+
+
+def make_sheet_name(template_name: str, type_: str):
+    """
+    Helper to create a Excel sheet name from the template name and it's type
+    (e.g. Fields or Template)
+
+    Create a sheet_name by putting together the template name and type. If
+    its > the max char limit for sheet names, truncate the template name
+    until the sheet_name has a valid length
+    """
+    suffix = f" - {type_}"
+    sheet_name = f"{template_name}{suffix}"
+    # Truncate template name to avoid Excel writer exception
+    if len(sheet_name) > MAX_SHEET_NAME_LEN:
+        sheet_name = template_name[
+            0:(MAX_SHEET_NAME_LEN - len(sheet_name))
+        ] + suffix
+    return sheet_name
+
+
+def templates_to_excel_workbook(
+    template_versions: List[TemplateVersion],
+    filepath_or_buffer: Union[str, BytesIO] = None,
```

Comment:
`filepath_or_buffer` should be `Optional[Union[str, BytesIO]]` or `Union[str, BytesIO, None]`